ts/components/messages/paginated/MessageList/Item.tsx

```diff
@@ -104,39 +104,89 @@ const styles = StyleSheet.create({
   }
 });
 
-function tagOrIcon({
+/**
+ * The data structure representing a badge.
+ */
+enum ItemBadge {
+  Paid,
+  Archived,
+  QrCode
+}
+
+/**
+ * Given various item states, return the corresponding
+ * visible badge.
+ */
+function getMaybeItemBadge({
```

Comment:
```suggestion
const getMaybeItemBadge = () => 
```
in the code base is widely used this convention to create a function